### PR TITLE
feat: fastify/express accepts an instance or instance factory

### PR DESCRIPTION
**Pull request by ghost**
Allow to pass a fastify/express instance factory to the module options.
Creating a new instance for every registration/init is safer to avoid registering plugins to the instance multiple times.

### CODE REVIEW

File: packages/http-server-fastify/src/FastifyHttpServer.ts [hunk 1]
```diff
@@ -27,10 +27,8 @@ import qs from 'qs';
 type Server = HttpServer | HttpsServer;
 
 export interface FastifyHttpServerModuleOptions extends HttpServerModuleOptions {
-	instance?: FastifyInstance;
-	middlewares?: {
```
Comment (pierissimo) [inline packages/http-server-fastify/src/FastifyHttpServer.ts:31]: keeping this nesting could be beneficial to maintain a similar structure as the [ExpressHttpServer module](https://github.com/HPInc/davinci/blob/master/packages/http-server-express/src/ExpressHttpServer.ts#L24).

Also, we should be careful not to introduce breaking changes too often

File: packages/http-server-fastify/src/FastifyHttpServer.ts [hunk 2]
```diff
@@ -27,10 +27,8 @@ import qs from 'qs';
...
-		cors?: FastifyCorsOptions;
-	};
+	instanceFactory?: () => FastifyInstance;
```
Comment (pierissimo) [inline packages/http-server-fastify/src/FastifyHttpServer.ts:30]:
```suggestion
	instance?: FastifyInstance | () => FastifyInstance;
```

**Review by pierissimo — CHANGES_REQUESTED** [review]

File: packages/http-server-fastify/src/FastifyHttpServer.ts (hunk 1 above)
Comment (adriasala124) [reply packages/http-server-fastify/src/FastifyHttpServer.ts:31]: My idea was to remove the middleware concept from fastify

**Review by pierissimo — APPROVED** [review]

File: packages/http-server-express/test/unit/ExpressHttpServer.test.ts [hunk 3]
```diff
@@ -60,6 +60,34 @@ describe('ExpressHttpServer', () => {
 		});
 	});
 
+	describe('instantiation', () => {
+		it('should allow to pass express instance factory in module options', async () => {
+			const instance = express();
+			// @ts-ignore
```
Comment (pierissimo) [inline packages/http-server-express/test/unit/ExpressHttpServer.test.ts:66]: are those @ts-ignore needed?

**Pull request merged**